otherlibs/systhreads/st_stubs.c

```diff
@@ -437,7 +437,8 @@ void caml_thread_interrupt_hook(void)
    system. */
 CAMLprim value caml_thread_initialize(value unit)
 {
-  CAMLparam0();
+  /* Protect against repeated initialization (PR#3532) */
+  if (Active_thread != NULL) return Val_unit;
```

Comment:
Why can we assume that `Active_thread` is NULL before initialization, and not some uninitialized value whose read is undefined behavior? `thread_table` does not seem 0-initialized.

Reply:
The “`thread_table` table” is actually 0-initialized, since it is a non-local static and an array of values.

Reply:
To me all fields should be indeed zero initialized as @gadmm mentioned, so I think this bit is ok. (`thread_table[n].active_thread` should indeed be 0)